trikNetwork/src/mailboxServer.cpp

```diff
@@ -124,38 +122,30 @@ void MailboxServer::connect(const QString &ip, int port)
 		return;
 	}
 
-	connect(server, serverPort);
+	connectTo(server, serverPort);
 }
 
-void MailboxServer::connect(const QString &ip)
+void MailboxServer::connectTo(const QString &ip)
 {
-	connect(ip, mMyPort);
+	connectTo(ip, mMyPort);
 }
 
-Connection *MailboxServer::connect(const QHostAddress &ip, int port)
+Connection *MailboxServer::connectTo(const QHostAddress &ip, int port)
 {
-	const auto connection = new MailboxConnection();
-
-	connectConnection(connection);
-
-	startConnection(connection);
-
-	QMetaObject::invokeMethod(connection, "connect"
-			, Q_ARG(QHostAddress, ip)
-			, Q_ARG(int, port)
-			, Q_ARG(int, mMyPort)
-			, Q_ARG(int, mHullNumber)
-			);
-
-	return connection;
+	const auto c = new MailboxConnection();
+	connectConnection(c);
```

Comment:
Who will be responsible for deleting it?

Reply:
`startConnection`  steals ownership and transfers it to a worker thread